src/LuaTransformer.ts

```diff
@@ -4167,6 +4229,56 @@ export class LuaTransformer {
         }
     }
 
+    public transformLuaTableCallExpression<T extends boolean>(
+        expression: ts.CallExpression,
+        isWithinExpressionStatement: T
+    ): T extends true ? tstl.Statement : tstl.Expression;
+    public transformLuaTableCallExpression(
+        expression: ts.CallExpression,
+        isWithinExpressionStatement: boolean
+    ): tstl.Statement | tstl.Expression | undefined {
+        const method = expression.expression as ts.PropertyAccessExpression;
+        const methodName = method.name.escapedText;
+        const signature = this.checker.getResolvedSignature(expression);
+        const tableName = (method.expression as ts.Identifier).escapedText;
+        const luaTable = tstl.createIdentifier(tableName);
+        const params = this.transformArguments(expression.arguments, signature);
+
+        switch (methodName) {
+            case "get":
+                if (expression.arguments.length !== 1) {
+                    throw TSTLErrors.ForbiddenLuaTableUseException(
+                        expression,
+                        "One parameter is required for get() on a '@LuaTable' object."
+                    );
+                }
+                if (isWithinExpressionStatement) {
```

Comment:
I don't think I understand the reasoning here. If you have a `table.get(abc)` statement it is transformed to `____ = table.get(abc)`? That sounds like asking for all kinds of strange bugs.

Reply:
Since `table.get(abc)` transforms to `table[abc]` it becomes an invalid expression in Lua. In this case I chose to then assign this to an anonymous variable the same way `x;` is transformed to `local ____ = x`;

Reply:
This sounds like a more fundamental issue. Table.get(abc) is an expression but you are returning a statement. This is related to my other comment on the function signature, I think it might automatically be resolved if you separate the assignment and expression functions.

Reply:
I've separated this logic into three functions atm to keep this logic more consistent. I've grouped the error checking in `validateLuaTableCall` which takes `isWithinExpressionStatement` for validation purposes, `transformLuaTableExpressionStatement` and `transformLuaTableExpression`.

Would this be something more appropriate?

Reply:
Seems okay, these methods should probably be private instead of public though.